src/state/login-form/actions.js

```diff
@@ -21,10 +20,16 @@ const ERROR_LOGIN_MESSAGE = 'error: username or password is invalid';
 export const performLogin = (username, password) => (dispatch) => {
   if (username && password) {
     dispatch(setLoginErrorMessage(''));
-    login(username, password).then(() => {
-      gotoRoute('dashboard');
+    return login(username, password).then((response) => {
+      if (response.ok) {
+        response.json().then(() => {
+          gotoRoute('dashboard');
```

Comment:
I know it was already like this, but could you use the constant `ROUTE_DASHBOARD` from `app-init/router` here?

Reply:
sure thing!